src/zkp/piaffg.rs

```diff
@@ -604,12 +608,437 @@ mod tests {
         with_random_paillier_affg_proof(rng, x, y, f)
     }
 
+    #[test]
+    fn piaffg_proof_with_consistent_secret_inputs_out_of_range() -> Result<()> {
+        let mut rng = init_testing();
+        let x_upper_bound = BigNumber::one() << (ELL + EPSILON);
+        let y_upper_bound = BigNumber::one() << (2 ^ (ELL_PRIME + EPSILON));
```

Comment:
Oh, I see I was wrong here because I misunderstood how these were being used. I thought they were the sampling bounds for the valid inputs, but they're the minimum bounds for invalid inputs.

Please tell me when you think I'm wrong about something like this.